maas/client/viscera/machines.py

```diff
@@ -190,18 +195,70 @@ class Machine(Object, metaclass=MachineType):
         if comment is not None:
             params["comment"] = comment
         data = await self._handler.deploy(**params)
-        return type(self)(data)
+        if wait is None:
+            return type(self)(data)
+        else:
+            # Wait for the machine to be fully deployed
+            machine = type(self)(data)
+            while machine.status == NodeStatus.DEPLOYING:
+                await asyncio.sleep(wait)
+                data = await self._handler.read(system_id=self.system_id)
+                machine = type(self)(data)
+            return machine
```

Comment:
I thought about that - yes I suppose it would make it more obvious to the user that something went wrong if we threw an exception.